packages/@ember/routing/route.ts

```diff
@@ -1825,6 +1826,15 @@ export function getRenderState(route: Route): RenderState | undefined {
   return route[RENDER_STATE];
 }
 
+import { precompileTemplate } from '@ember/template-compilation';
+
+function RouteTemplate(Component: object): TemplateFactory {
+  return precompileTemplate(`<Component @model={{@model}} @controller={{this}} />`, {
+    strictMode: true,
+    scope: () => ({ Component }),
+  });
+}
```

Comment:
This is.. fine.. I guess. I don't have time to dig deeper but it feels unnecessary. Somewhere not that much deeper (inside this repository I am pretty sure), we are interfacing with the low-level APIs in glimmer-vm that lets us render arbitrary template with arbitrary `this` arg and named arguments, so this extra indirection sticks out to me

Reply:
This may also not be fine, actually. It at least _looks_ like this could potentially leak and be slow. Making and returning a new template each time this is likely problematic, as each unique template needs to be "runtime compiled" which is, IIRC, not-GC-able, at least not until the owner is destroyed.

Reply:
Basically – `@model` is already a magic argument ember adds at this integration point. So we must already be very close to the place where we can inject any number of these extra arguments. We should just be able to add `@controller` the same way.

If we _must_ have a wrapper template, I think we should be able to do a single static template and pass `@component` as an argument, rather than building dynamically building a new one each time.

Reply:
I implemented this suggestion so now the template is stable and the component is parameterized.